pkg/config/utils/telemetry.go

```diff
@@ -32,5 +37,11 @@ func IsCheckTelemetryEnabled(checkName string, cfg pkgconfigmodel.Reader) bool {
 
 // IsTelemetryEnabled returns whether or not telemetry is enabled
 func IsTelemetryEnabled(cfg pkgconfigmodel.Reader) bool {
-	return cfg.IsSet("telemetry.enabled") && cfg.GetBool("telemetry.enabled")
+	return cfg.IsSet("telemetry.enabled") && cfg.GetBool("telemetry.enabled") ||
+		(IsAgentTelemetryEnabled(cfg))
+}
+
+// IsAgentTelemetryEnabled returns whether or not agent telemetry is enabled
+func IsAgentTelemetryEnabled(cfg pkgconfigmodel.Reader) bool {
```

Comment:
Every time we make a public function on this package tends to be use in many places 😢 Ideally in the future we would have a single configuration component 🎉 

Since this function is only use here could we make it private or inline it?

Reply:
Right, made it private. Ideally we would like to join these two configurations in the future someway. @vickenty suggested to use a slightly different naming to indicate similar lineage (see his comment above)